src/printer.js

```diff
@@ -1996,6 +2004,39 @@ function printArgumentsList(path, options, print) {
     ]);
   }
 
+  if (shouldGroupFirstArg(args)) {
+    const shouldBreak = printed.slice(1).some(willBreak);
+    return concat([
+      printed.some(willBreak) ? breakParent : "",
+      conditionalGroup(
+        [
+          concat(["(", join(concat([", "]), printed), ")"]),
+          concat([
+            "(",
+            group(printed[0], { shouldBreak: true }),
```

Comment:
it looks like this part is the only difference. Could you try to merge the two instead of copy pasting everything?

Something like that (feel free to find better variable names:

```js
const groupFirst = shouldGroupFirst()
const groupLast = shouldGroupLast()

if (groupFirst || groupLast) {
  const begin = groupFirst ? printed[0] : printed.slice(0, -1);
  const end = groupFirst ? printed.slice(1) : util.getLast(printed);

  // ...
}
```